Source/WTF/wtf/Compiler.h

```diff
@@ -270,11 +270,15 @@
 
 /* MUST_TAIL_CALL */
 
+// 32-bit platforms use different calling conventions, so a MUST_TAIL_CALL function
+// written for 64-bit may fail to tail call on 32-bit.
+#if !(COMPILER(CLANG) && __SIZEOF_POINTER__ == 4)
```

Comment:
I think, because of how things are handled in preprocessor, it should be,
```
#if COMPILER(CLANG)
#if __SIZEOF_POINTER__ == 8
...
#endif
#endif
```